test/proper_lib.erl

```diff
@@ -12,11 +12,23 @@
 codec(Mod, Opts, Data) ->
     codec(Mod, Opts, Data, fun(V) -> V end).
 
-codec(Mod, Opts, Data, Canonical) ->
+% Encode and then decode the Input
+encode_decode(Mod, Opts, Input) ->
     {_, Config} = Mod:init(Opts),
     TypeInfo = #type_info{config = Config},
-    <<Size:32, Encoded:Size/binary>> = iolist_to_binary(Mod:encode(Data, TypeInfo)),
-    Canonical(Data) =:= Canonical(Mod:decode(Encoded, TypeInfo)).
+    <<Size:32, Encoded:Size/binary>> = iolist_to_binary(Mod:encode(Input, TypeInfo)),
+    Mod:decode(Encoded, TypeInfo).
```

Comment:
I found the unit tests more readable with `?assertEqual` and therefore it is useful to simply have the value back after encoding and then decoding.